src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -23,6 +23,60 @@ class Lambda {
   private val cloudWatchClient = AmazonCloudWatchClientBuilder.defaultClient
   private val snsClient = AmazonSNSClientBuilder.defaultClient
 
+  private def raiseAllThePurges(event: Event): Boolean = {
+    val contentType = extractUpdateContentType(event)
+    val purgeType = if (event.eventType == EventType.Delete) {
+      Hard
+    } else {
+      Soft
+    }
+    val aliasPaths = extractAliasPaths(event)
+    val dotcomCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, purgeType, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey, contentType)
+    val jsonCanonicalPurge = sendFastlyPurgeRequestForAjaxFile(event.payloadId, contentType)
+    val mapiCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, purgeType, config.fastlyMapiServiceId, makeMapiSurrogateKey(event.payloadId), config.fastlyMapiApiKey, contentType)
+    val aliasPurges = aliasPaths.fold(Seq.empty[Boolean])({
```

Comment:
To remove duplication, could possibly create a `pathsToPurge` variable which combines `aliasPaths` and `event.payloadId` and fold over that instead?

Reply:
I believe I've now done something like this; building a list of paths and then decaching them.